lib/coverband/configuration.rb

```diff
@@ -7,7 +7,8 @@ class Configuration
                   :reporter, :redis_namespace, :redis_ttl,
                   :background_reporting_enabled,
                   :background_reporting_sleep_seconds, :test_env,
-                  :web_enable_clear, :gem_details, :web_debug, :report_on_exit
+                  :web_enable_clear, :gem_details, :web_debug, :report_on_exit,
+                  :simulate_oneshot_lines_coverage
```

Comment:
Could we instead just use the configuration `use_oneshot_lines_coverage` and then only simulate if ruby version does not support one shot line coverage?

Reply:
yeah I did consider that, but there are some tradeoffs, so I wanted to make it a bit more explicit... basically while it doesn't store anything in memory between requests it does have a bit higher allocations.

Reply:
One thought. Was under the impression that peek_results is what causes the memory bloat. If that is the case, why would simulate_oneshot_coverage help since we are still calling peek_results?

Reply:
It removes storing any memory between requests... so there is no `@@previous_coverage` holding a large JSON structure... Basically, those were the two biggest issues

*  `peek_results` itself can cause memory bloat if you end up in a base situation when we have to malloc and GC memory all the time... Can't fix that until `2.6.0`
* We significantly increase memory all the time if we have to keep `@@previous_coverage` in memory between requests because then we can't even GC it. (simulated oneshot or real oneshot, fixes this issue)

I guess the 3rd issue is what you have been trying to fix, which is that our serialization just has far to large of memory cost.

Make sense @kbaum ?

Reply:
Makes sense. Thanks for explanation!